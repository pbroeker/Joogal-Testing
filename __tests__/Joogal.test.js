import React from 'react';
import Joogal from '../Joogal';
import { fireEvent, render } from '@testing-library/react-native';

const user = {
  id: 3,
  artistname: 'Kollwitz',
  firstname: 'kaethe',
  lastname: 'kollwitz',
  email: 'kaethe@artists.de',
  roles: ['painter'],
  about: 'a German artist who worked with painting, printmaking and sculpture',
  likes: [{ target: 1 }],
  profilepic: null
};

const artist1 = {
  id: 1,
  likes: [{ target: 3 }],
  artistname: 'Kubrick',
  firstname: 'Stanley',
  lastname: 'Kubrick',
  email: 'kubrick@artists.com',
  roles: ['filmmaker'],
  about: 'American film director',
  profilepic: null
};

const artist2 = {
  likes: [{ target: 1 }],
  profilepic: null,
  id: 2,
  artistname: 'Eminem',
  firstname: 'Marshall',
  lastname: 'Mathers',
  email: 'eminem@artists.com',
  roles: ['rapper'],
  about: 'is an American rapper, songwriter, and record producer'
};

const allArtists = [artist1, artist2];
const refreshUser = jest.fn();

beforeEach(() => {
  joogalInstance = render(<Joogal user={user} allArtists={allArtists} refreshUser={refreshUser} />);
});

afterEach(() => {
  joogalInstance.unmount();
});

test('component mounts', () => {
  expect(joogalInstance.getByTestId('Joogal')).toBeTruthy();
});

test('renders correctly', () => {
  const text = joogalInstance.getByText('Collaborations');
  expect(text).toBeTruthy();
});

test('displays liked artists', () => {
  const liked = joogalInstance.getByText(`${artist1.firstname} ${artist1.lastname}`);
  expect(liked).toBeTruthy();
});

test('does not display artists that are not liked', () => {
  const notLiked = joogalInstance.queryByText(`${artist2.firstname} ${artist2.lastname}`);
  expect(notLiked).toBeNull();
});

test('snapshot', () => {
  expect(joogalInstance.toJSON()).toMatchSnapshot();
});

test('should refresh', () => {
  const flatList=joogalInstance.getByTestId('Joogal-FlatList');
  expect(refreshUser).toHaveBeenCalledTimes(0);
  fireEvent(flatList.props.refreshControl, 'refresh');
  expect(refreshUser).toHaveBeenCalledTimes(1);
});