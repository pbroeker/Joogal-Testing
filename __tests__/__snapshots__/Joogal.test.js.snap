// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1e20",
      "height": "100%",
    }
  }
  testID="Joogal"
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#1d1e20",
          "color": "#f7f7f7",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 24,
          "letterSpacing": 8,
          "marginTop": 251,
        },
        Object {
          "marginTop": 30,
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Collaborations
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderColor": "#f7f7f7",
          "borderWidth": 0.5,
          "marginTop": 298,
          "position": "absolute",
          "width": "95%",
        },
        Object {
          "marginTop": 100,
        },
      ]
    }
  />
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "about": "American film director",
          "artistname": "Kubrick",
          "email": "kubrick@artists.com",
          "firstname": "Stanley",
          "id": 1,
          "lastname": "Kubrick",
          "likes": Array [
            Object {
              "target": 3,
            },
          ],
          "profilepic": null,
          "roles": Array [
            "filmmaker",
          ],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
        tintColor="whitesmoke"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 24,
      }
    }
    testID="Joogal-FlatList"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "backgroundColor": "#f7f7f7",
                  "borderColor": "#f7f7f7",
                  "borderRadius": 100,
                  "borderWidth": 1,
                  "height": 100,
                  "marginLeft": 10,
                  "marginVertical": 5,
                  "width": 100,
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#f7f7f7",
                  "fontFamily": "PoiretOne_400Regular",
                  "fontSize": 16,
                  "letterSpacing": 5,
                  "marginLeft": 10,
                  "textTransform": "uppercase",
                  "top": 10,
                }
              }
            >
              Stanley
               
              Kubrick
            </Text>
            <Text
              style={
                Object {
                  "color": "#f7f7f7",
                  "fontFamily": "PoiretOne_400Regular",
                  "fontSize": 16,
                  "letterSpacing": 5,
                  "marginLeft": 10,
                  "textTransform": "uppercase",
                  "top": 10,
                }
              }
            >
              Kubrick
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#f7f7f7",
                    "fontFamily": "PoiretOne_400Regular",
                    "fontSize": 14,
                    "letterSpacing": 1,
                    "marginLeft": 10,
                    "top": 18,
                    "width": "55%",
                  },
                  Object {
                    "color": "lightgreen",
                    "fontSize": 19,
                    "marginLeft": 50,
                    "width": 400,
                  },
                ]
              }
            >
              kubrick@artists.com
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
